pandas/tests/io/test_sql.py

```diff
@@ -565,14 +620,28 @@ def test_dataframe_to_sql_arrow_dtypes(conn, request):
             "string": pd.array(["a"], dtype="string[pyarrow]"),
         }
     )
+
     conn = request.getfixturevalue(conn)
+
     with tm.assert_produces_warning(UserWarning, match="the 'timedelta'"):
         df.to_sql("test_arrow", conn, if_exists="replace", index=False)
 
 
 @pytest.mark.db
 @pytest.mark.parametrize("conn", all_connectable)
 def test_dataframe_to_sql_arrow_dtypes_missing(conn, request, nulls_fixture):
+    if conn == "postgresql_adbc_conn":
+        request.node.add_marker(
+            pytest.skip("int8/datetime not implemented yet in adbc driver")
```

Comment:
There is no such thing as an 8-bit integer in postgres, so this test wouldn't round trip in its current form. I think this being an int8 is just a test detail and not anything we actually care for. Can probably bump to int16, though that separately begs the question of how we want to add/extend the ADBC types

Reply:
Whats the canonical way to add ADBC types?

Reply:
Are you referring to the mapping of an arrow type to a database-specific type?

Reply:
Yes

Reply:
That could in theory occur in the driver or could be handled by pandas itself. Since timestamp is a primitive in most databases I think that will be handled by the driver (plan on looking at this today/tomorrow myself for postgres)

Int8 is a little trickier. From the databases I've used I am not aware of 8 bit integers being all that common, so either the driver implements some compatability to another type (logically int16) or leaves it to the application to put its data into a supported type. My guess is the ADBC drivers may want to start explicitly and force end users to upcast to int16 if that's what they want, but @lidavidm knows best

Reply:
If there's a strong convention (like timestamp strings in SQLite) I'd like to support it natively, otherwise I think it would be best for the client to be explicit about what it wants (and if there's something the driver can do to help, we can do it - e.g. I think it's reasonable to optionally ingest int8 as SMALLINT in the driver, so long as you aren't concerned about perfect roundtripping)

Reply:
Well we kinda do this already for numpy types -> sqlalchemy types so not opposed to adding that in pandas. I was just curious what the API for defining these is like.

Reply:
int8 support was added in https://github.com/apache/arrow-adbc/pull/858 and working on timestamp in https://github.com/apache/arrow-adbc/pull/861; guessing these can make it into the next ADBC release and we can just start support at that as a minimum version to keep things easy